modules/RaptMediaNg/resources/mw.RaptMediaNg.js

```diff
@@ -0,0 +1,508 @@
+/**
+ * The RaptMediaNg plugin integrates the RaptMedia Engine to the Kaltura Player.
+ * RaptMedia adds clickable interactive layer that accompanies your video content and can do things like:
+ * cue or launch different media plays, jump to specific timecode, trigger an event on your webpage and launch a new web page or an app.
+ * This plugin also makes use of accompanying plugin RaptMediaNgScrubber plugin to override the default scrubber behavior to fit a Rapt Media experience.
+ * With RaptMediaNgScrubber plugin the scrubber can interact within the context of a single RaptMedia clip instead of just the entire stitched playlist.
+ * The RaptMediaNg plugin integrates the RaptMedia Engine to the Kaltura Player.
+ * It also makes use of accompanying plugin RaptMediaNgDurationLabel used to override the default player DurationLabel to behave according to the RaptMedia Sequence rather than show the overall playlist duration.
+ */
+(function ( mw, $ ) {
+	"use strict";
+
+	var AbortError = function() {};
+
+	mw.PluginManager.add( 'raptMediaNg', mw.KBaseComponent.extend( {
+
+		defaultConfig: {
+			parent: 'videoHolder'
+		},
+
+		setup: function(){
+			this.initialize();
+			this.addBindings();
+		},
+
+		initialize: function() {
+			this.setConfig('status', 'disabled', true);
+
+			this.setConfig('projectId', undefined, true);
+			this.setConfig('info', undefined, true);
+
+			this.pendingEntryId = null;
+			this.targetEntryId = null;
+		},
+
+		addBindings: function() {
+			var _this = this;
+
+			this.bind('raptMediaNg_doPlay', function(event) {
```

Comment:
Since these bindings (`raptMedia_doPlay`, `raptMedia_doPause`, etc) are primarily designed to support custom external API usage, I would recommend leaving the prefix as `raptMedia_` instead of changing it to `raptMediaNg_`. That way it will be easier to switch between the two plugins without having to update any custom API usage the user may have external to the player.

I do realize this creates some room for confusion, particularly since there are a few events that are only indented to be internal. As such I would probably recommend sticking with a `raptMedia_` prefix for all events.

Reply:
Hi @khalsah - 

First of all, thanks for reviewing.
Generally speaking, I concur [it will be hard for anyone to argue with this logic, IMHO].
However, because of the nature of this particular delivery, I'd rather keep things as they are for now. If we end up providing this to additional customers and the plugin is recognised as an officially supported one, I will be glad to do some re-factoring [shouldn't take more than 5 minutes] and let it run through a proper QA process.

The above true to all your other comments as well.